cmd/opampsupervisor/supervisor/commander/commander_windows.go

```diff
@@ -6,25 +6,46 @@
 package commander
 
 import (
+	"fmt"
 	"os"
 	"syscall"
 
 	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/svc"
 )
 
 var (
 	kernel32API = windows.NewLazySystemDLL("kernel32.dll")
 
-	ctrlEventProc = kernel32API.NewProc("GenerateConsoleCtrlEvent")
+	ctrlEventProc    = kernel32API.NewProc("GenerateConsoleCtrlEvent")
+	allocConsoleProc = kernel32API.NewProc("AllocConsole")
+	freeConsoleProc  = kernel32API.NewProc("FreeConsole")
 )
 
 func sendShutdownSignal(process *os.Process) error {
 	// signaling with os.Interrupt is not supported on windows systems,
 	// so we need to use the windows API to properly send a graceful shutdown signal.
 	// See: https://learn.microsoft.com/en-us/windows/console/generateconsolectrlevent
+	isService, err := svc.IsWindowsService()
+	if err != nil {
+		return fmt.Errorf("isWindowsService: %w", err)
+	}
+
+	if isService {
+		if err := allocConsole(); err != nil {
+			return fmt.Errorf("allocConsole: %w", err)
+		}
+	}
```

Comment:
Honestly I'm not totally sure. From what I could find, because windows services aren't attached to a console, it's best to use them only when needed. The idea of a service is it can be ran in the background without user input, but attaching a console to one conflicts with that, so I guess it's better to only use it when strictly necessary.

Reply:
Hmm, it seems more error prone to alloc/free the console on every restart. For instance, if the free fails, then alloc will never succeed, so the agent can never be stopped gracefully in this function.

It also probably simplifies the code a bit since we wouldn't need to pass through whether this is a windows service or not (as per Paulo's suggestion).

I think I'd prefer just allocating it once up front unless we are observing some negative side-effect.